### DISCUSSION TITLE
Create node set from post adaptivity mesh failure

### DISCUSSION THREAD

**Discussion by LifengGan** [idaholab/moose#27707] [Q&A Meshing] [answered]
Dear moose user,

I had a problem on making node set from the post-adaptivity mesh using the following Mesh and Adaptivity block

I got the noncrack node set only the the original gen mesh, not the one from post adaptivity, see the fig below when loading in y-direction condition

Would anyone know how to fix this problem, thx.

Best regards,
Lifeng

```

[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 20
    ny = 10
    xmax = 0.1
    ymax = 0.05
  []
  [./noncrack]
    type = BoundingBoxNodeSetGenerator
    new_boundary = noncrack
    bottom_left = '0.05 0 0'
    top_right = '0.1 0 0'
    input = gen
  [../]
[]

[Adaptivity]
  marker = marker
  initial_marker = marker
  initial_steps = 2
  stop_time = 0
  max_h_level = 5
  [Markers]
    [marker]
      type = OrientedBoxMarker
      center = '0.07 0 0'
      length = 0.055
      width = 0.02
      height = 0.1
      length_direction = '1 0 0'
      width_direction = '0 1 0'
      inside = REFINE
      outside = DO_NOTHING
    []
  []
[]

```

![output_image_0](https://github.com/idaholab/moose/assets/69451718/52dcc8c8-c30c-4109-aa92-98ab25310ff6)

**Comment by GiudGiud** [comment, marked as answer]
Hello

The ordering of execution does not work to create the nodeset like this. 
The mesh is generated by `gen` and `noncrack`, then adaptivity is run.

You could:
- save the mesh using the checkpoint system (.cpr files), reload it with a FileMeshGenerator, then run the `noncrack` generator again
OR
- try to create a new sideset-updater user object that looks like BoundingBoxNodesetGenerator
An example here
https://mooseframework.inl.gov/source/userobjects/SidesetAroundSubdomainUpdater.html
of the user-object version of the SidesetAroundSubdomainGenerator

Guillaume
